File: a824(treasure)/C++/treasure.cpp
#include <iostream>
#include <algorithm>

int func(int n, int ceil) {
  if (n > ceil) return 0;
  int h = ceil / n;
  return (n + n * h) * h * 0.5;
}

char alphabet[27] = {'0', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', \
            'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'};

int main(void) {
  int a, b, c, n;
  while (std::cin >> a >> b >> c) {
    n = func(a, c) + func(b, c) - func(a * b / std::__gcd(a, b), c);
    if (n > 26) n %= 26;
    std::cout << alphabet[n] << '\n';
  }
  return 0;
}

File: a824(treasure)/Python/treasure.py
from sys import stdin, stdout
from math import gcd
alphabet = ['0', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', \
            'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z']
def func(n: int, ceil: int) -> int:
    if n > ceil: return 0
    h = ceil // n
    return ((n * h + n) * h) // 2
while True:
    line = stdin.readline()
    if not line: break
    a, b, c = map(int, line.split())
    n = func(a, c) + func(b, c) - func(a * b // gcd(a, b), c)
    if n > 26: n %= 26
    stdout.write(alphabet[n] + '\n')
